test/bats/azure.bats

```diff
@@ -369,7 +369,11 @@ setup() {
 }
 
 @test "Test filtered-watch-secret=false for nodePublishSecretRef" {
-  run helm upgrade csi-secrets-store manifest_staging/charts/secrets-store-csi-driver --reuse-values --set filteredWatchSecret=false --wait --timeout=5m -v=5 --debug --namespace kube-system
+  local chart_dir=${HELM_CHART_DIR}
+  if [[ "${chart_dir}" == "" ]]; then
+    chart_dir=manifest_staging/charts/secrets-store-csi-driver
+  fi
+  run helm upgrade csi-secrets-store "${chart_dir}" --reuse-values --set filteredWatchSecret=false --wait --timeout=5m -v=5 --debug --namespace kube-system
```

Comment:
since we are doing `make e2e-bootstrap e2e-helm-deploy-release e2e-azure && HELM_CHART_DIR=manifest_staging/charts/secrets-store-csi-driver make e2e-helm-upgrade e2e-azure` in test-infra does that mean we cleanup everything before testing upgrade? If so, then where are we doing in-place upgrade tests to determine how new release is interacting with existing secrets store resources?

Reply:
Hey @ritazh, We are not doing cleanup. `e2e-bootstrap` sets Kind. Then we use `e2e-helm-deploy-release` to install latest released chart and run e2e tests against it. Then `e2e-helm-upgrade` upgrades the chart from _manifest_staging/charts/secrets-store-csi-driver_ (aka next version/chart to be released) and run e2e tests again to validate upgrade.

P.S. Test being referred here is removed. We have made `filtered-watch-secret=true` as default from `v0.1.0` and this flag will be deprecated in `v1.0.0`. (Periodic job in test-infra still test upgrade with other tests.)

Reply:
@nilekhc By cleanup I mean we delete all the test resources from the cluster in teardown_file. So we don't really test how the upgraded component works with existing resources.

Reply:
Got it. Let me open a PR to add inplace upgrade tests as well. Thanks for reviewing!

Reply:
Created https://github.com/kubernetes-sigs/secrets-store-csi-driver/issues/739

Reply:
Implemented via https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/741